### PR TITLE
fix: Set Current Surface as first surface for direct navigator

### PR DESCRIPTION
**Pull request by CarloVarni**
For a direct navigator we set the current surface to be the first surface in the surface sequence. This is done in the `DirectNavigator:Initializer::operator()`

### CODE REVIEW

File: Core/include/Acts/Propagator/DirectNavigator.hpp [hunk 1]
```diff
@@ -72,6 +72,7 @@ class DirectNavigator {
       if (not r.initialized) {
         // Initialize the surface sequence
         state.navigation.navSurfaces = navSurfaces;
+        state.navigation.currentSurface = navSurfaces.front();
```
Comment (paulgessinger) [inline Core/include/Acts/Propagator/DirectNavigator.hpp:75]: Maybe we should check if we are on the first surface before doing this? I'm not sure the start position is always guaranteed to be on the first surface.

Comment (andiwand) [reply Core/include/Acts/Propagator/DirectNavigator.hpp:75]: In `Navigator.hpp` we do this during initialization https://github.com/acts-project/acts/blob/main/Core/include/Acts/Propagator/Navigator.hpp#L326

I think we should stick to this pattern here

Comment (CarloVarni) [reply Core/include/Acts/Propagator/DirectNavigator.hpp:75]: makes sense ... indeed the `initialize()` method for the `DirectNavigator` does nothing

Comment (CarloVarni) [reply Core/include/Acts/Propagator/DirectNavigator.hpp]: concerning check if we are on the first surface ... is there a way of retrieving that info from the state?

File: Core/include/Acts/Propagator/DirectNavigator.hpp [hunk 2]
```diff
@@ -12,6 +12,7 @@
 #include "Acts/Geometry/Layer.hpp"
 #include "Acts/Geometry/TrackingGeometry.hpp"
 #include "Acts/Geometry/TrackingVolume.hpp"
+#include "Acts/Navigation/DetectorNavigator.hpp"
```
Comment (andiwand) [inline Core/include/Acts/Propagator/DirectNavigator.hpp:15]:
```suggestion
```

**Review by andiwand — COMMENTED** [review]
one minor thing then ready to go - thanks!

File: Core/include/Acts/Propagator/DirectNavigator.hpp (hunk 2 above)
Comment (CarloVarni) [reply Core/include/Acts/Propagator/DirectNavigator.hpp:15]: I had to add it due to SYCL complaining on `volInfo`

Comment (andiwand) [reply Core/include/Acts/Propagator/DirectNavigator.hpp:15]: oh I see the function is not defined here but in the other classes / headers. I think we can copy paste it into the private section

from `Navigator.hpp`

```
  template <typename propagator_state_t>
  std::string volInfo(const propagator_state_t& state) const {
    return (state.navigation.currentVolume
                ? state.navigation.currentVolume->volumeName()
                : "No Volume") +
           " | ";
  }
```

Comment (CarloVarni) [reply Core/include/Acts/Propagator/DirectNavigator.hpp]: done. Maybe in future MRs we can make this a global function instead

**Review by andiwand — APPROVED** [review]

**Comment by CarloVarni** [comment]
new commit for fixing format. @andiwand need another approval

**Review by andiwand — APPROVED** [review]

**Comment by acts-project-service** [comment]
## :white_check_mark: Athena integration test results [159acb7f2b1c631748c4ea2a37d1ead092bb69e7]
### :white_check_mark: All tests successful

|status|job|
|------|---|
|:green_circle: | [run_workflow_tests_run2_data](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33578340) |
|:green_circle: | [run_unit_tests](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33578038) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsBenchmarkWithSpot.sh 8 100](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33577592) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsWorkflow.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33577591) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateAmbiguityResolution.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33577590) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateTracks.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33577589) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateActsCoreSpacePoints.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33577588) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateActsSpacePoints.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33577587) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateSeeds.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33577586) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateOrthogonalSeeds.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33577585) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateClusters.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33577584) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsPersistifyEDM.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33577583) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsKfRefitting.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33577582) |
|:green_circle: | [run_ci_tests: python3 ../athena/Tracking/Acts/ActsGeometry/test/ActsExtrapolationAlgTest.py](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33577581) |
|:green_circle: | [run_ci_tests: python3 ../athena/Tracking/Acts/ActsGeometry/test/ActsITkTest.py](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33577580) |
|:green_circle: | [run_workflow_tests_run4_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33577579) |
|:green_circle: | [run_workflow_tests_run3_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33577577) |
|:green_circle: | [run_workflow_tests_run3_data](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33577576) |

**Pull request merged**